core/server/proxy/src/main/java/alluxio/proxy/s3/signature/AuthorizationV4HeaderParser.java

```diff
@@ -0,0 +1,161 @@
+      StringTokenizer tokenizer = new StringTokenizer(parsedSignedHeaders, ";");
+      if (tokenizer.hasMoreTokens()) {
+        return parsedSignedHeaders;
+      } else {
+        LOG.error("No signed headers found. AuthHeader:{}", mAuthHeader);
+        throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+      }
+    } else {
+      LOG.error("No signed headers found. AuthHeader:{}", mAuthHeader);
+      throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+    }
+  }
+
+  /**
+   * This method parses authorization header.
+   * @throws S3Exception
+   */
+  @Override
+  public SignatureInfo parseSignature() throws S3Exception {
+    if (mAuthHeader == null || !mAuthHeader.startsWith("AWS4")) {
+      return null;
+    }
+    int algorithmPos = mAuthHeader.indexOf(' ');
+    if (algorithmPos < 0) {
+      throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+    }
+
+    //split the value parts of the authorization header
+    String[] split = mAuthHeader.substring(algorithmPos + 1).trim().split(", *");
+
+    if (split.length != 3) {
+      throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+    }
+
+    String algorithm = mAuthHeader.substring(0, algorithmPos);
+    validateAlgorithm(algorithm);
+    AWSCredential credentialObj = parseCredentials(split[0]);
+    String signedHeaders = parseSignedHeaders(split[1]);
+    String signature = parseSignature(split[2]);
+    return new SignatureInfo(
+            SignatureInfo.Version.V4,
+            credentialObj.getDate(),
+            mDateHeader,
+            credentialObj.getAccessKeyID(),
+            signature,
+            signedHeaders,
+            credentialObj.createScope(),
+            algorithm,
+            true
+    );
+  }
+
+  /**
+   * Validate signature.
+   */
+  private String parseSignature(String signature) throws S3Exception {
+    if (signature.startsWith(SIGNATURE)) {
+      String parsedSignature = signature.substring(SIGNATURE.length());
+      if (StringUtils.isEmpty(parsedSignature)) {
+        LOG.error("Signature can't be empty: {}", signature);
+        throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+      }
+      try {
+        Hex.decodeHex(parsedSignature);
+      } catch (DecoderException e) {
+        LOG.error("Signature:{} should be in hexa-decimal encoding.", signature);
+        throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+      }
+      return parsedSignature;
+    } else {
+      LOG.error("No signature found: {}", signature);
+      throw new S3Exception(mAuthHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+    }
+  }
+
+  /**
+   * Validate credentials.
+  */
+  private AWSCredential parseCredentials(String credential) throws S3Exception {
+    AWSCredential credentialObj = null;
```

Comment:
This variable is not needed. The function can be written as
```
if (....) {
  return new AWSCredential(...);
}
throw new S3Exception(...);
```
Or as I suggested earlier
```
if (...) {
  return AWSCredential.createCredential(....);
}
throw ...
```